libglusterfs/src/dict.c

```diff
@@ -684,7 +684,28 @@ dict_deln(dict_t *this, char *key, const int keylen)
     return rc;
 }
 
-void
+/* removes and free all data_pair_t elements.
+ * has to be called with this->lock held */
+static void
```

Comment:
Here is the point. this->count gives you the number of pairs in this dictionary.
So while in a while loop we are destroying/freeing data pairs, we should also keep decreasing it.
Or at the end make it 0.

Reply:
Agree, will add this to the function.